src/plugins/intel_gpu/src/graph/permute.cpp

```diff
@@ -74,7 +74,7 @@ std::vector<layout> permute_inst::calc_output_layouts(permute_node const& /*node
         }
     }
 
-    for (int64_t i = 0; i < input_static_rank; ++i) {
+    for (size_t i = 0; i < permute_order.size(); ++i) {
```

Comment:
please add hadnling input layout is dynamic not only input rank is dynamic, because this accessing index crashes when the input layout is dynamic.